azurelinuxagent/ga/exthandlers.py

```diff
@@ -623,7 +633,7 @@ def decide_version(self, target_state=None):
         try:
             pkg_list = self.protocol.get_ext_handler_pkgs(self.ext_handler)
         except ProtocolError as e:
-            raise ExtensionError("Failed to get ext handler pkgs", e)
+            raise ExtensionDownloadError("Failed to get ext handler pkgs", e)
```

Comment:
get_ext_handler_pkgs() may raise ProtocolError for reasons other than download errors. I think you want to make the conversion to DownloadError deeper in the code, probably somewhere in the call to or implementation of get_ext_manifest

Reply:
Thanks for the tip. Went through the call tree. Will take another look to narrow down the exception causing areas.